### PR TITLE
Fix notifications id

### LINKED ISSUE #208: All notifications' settings belong to the other category
### What Happened?

Since 7.0.0 I've noticed I started receiving notifications for apps that I have disabled notifications for. They are still disabled in settings, but that seems to have no effect. However, changing settings for the Other category at the bottom applies to all notifications.
Tried toggling these settings on/off for apps and rebooting the device.

### Expected Behavior

Notifications settings should be applied on a per-app basis as before.

### OS Version

7.x (Horus)

### Software Version

Latest release (I have run all updates)

### PR DESCRIPTION
**Pull request by lenemter**
Fixes #208

### CODE REVIEW

File: src/Notification.vala [hunk 1]
```diff
@@ -84,8 +84,11 @@ public class Notifications.Notification : GLib.Object {
         if ("desktop-entry" in hints && hints["desktop-entry"].is_of_type (VariantType.STRING)) {
             app_info = new DesktopAppInfo ("%s.desktop".printf (hints["desktop-entry"].get_string ()));
 
-            if (app_info != null && app_info.get_boolean ("X-GNOME-UsesNotification")) {
-                app_id = app_info.get_id ();
+            if (app_info != null && app_info.get_boolean ("X-GNOME-UsesNotifications")) {
+                var app_info_id = app_info.get_id ();
+                if (app_info_id != null) {
+                    app_id = app_info_id.replace (".desktop", "");
```
Comment (tintou) [inline src/Notification.vala:90]: It's harder than that as there are apps like "com.nextcloud.desktopclient.nextcloud":
```suggestion
                    if (app_info_id.has_suffix (".desktop")) {
                        app_id = app_info_id.substring (0, - ".desktop".length)
                    } else {
                        app_id = app_info_id;
                    }
```

File: src/Notification.vala [hunk 2]
```diff
@@ -84,8 +84,15 @@ public class Notifications.Notification : GLib.Object {
         if ("desktop-entry" in hints && hints["desktop-entry"].is_of_type (VariantType.STRING)) {
             app_info = new DesktopAppInfo ("%s.desktop".printf (hints["desktop-entry"].get_string ()));
 
-            if (app_info != null && app_info.get_boolean ("X-GNOME-UsesNotification")) {
-                app_id = app_info.get_id ();
+            if (app_info != null && app_info.get_boolean ("X-GNOME-UsesNotifications")) {
+                var app_info_id = app_info.get_id ();
+                if (app_info_id != null) {
+                    if (app_info_id.has_suffix (".desktop")) {
+                        app_id = app_info_id.substring (0, app_info_id.length - ".desktop".length);
```
Comment (tintou) [inline src/Notification.vala:91]: > If offset is negative, string.substring uses it as a character index from the end of the string.
```suggestion
                        app_id = app_info_id.substring (0, - ".desktop".length);
```
so this should theoretically work

Comment (lenemter) [reply src/Notification.vala:91]: It doesn't.

Comment (tintou) [reply src/Notification.vala:91]: Oh yeah sorry you're right, that's offset not len

**Review by tintou — APPROVED** [review]

**Pull request merged**